src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliExecutor.cs

```diff
@@ -88,14 +90,16 @@ public ExecuteResult Execute(ExecuteParameters executeParameters)
 
                 loggerWithDiagnoser.ProcessInput();
 
-                if (!process.WaitForExit(milliseconds: 250))
+                if (!process.WaitForExit(milliseconds: (int)ProcessExitTimeout.TotalMilliseconds))
                 {
-                    logger.WriteLineInfo("// The benchmarking process did not quit on time, it's going to get force killed now.");
+                    logger.WriteLineInfo($"// The benchmarking process did not quit within {ProcessExitTimeout.TotalSeconds} seconds, it's going to get force killed now.");
 
                     consoleExitHandler.KillProcessTree();
+                    // Give KillProcessTree a few moments to actually kill the benchmarking process.
+                    Thread.Sleep(millisecondsTimeout: 500);
```

Comment:
`KillProcessTree` calls `KillTree`
https://github.com/dotnet/BenchmarkDotNet/blob/63e28c100a42a6492d09a0b93a8a4c141061bb0d/src/BenchmarkDotNet/Helpers/ConsoleExitHandler.cs#L60-L66

which uses the default timeout of 30s:
https://github.com/dotnet/BenchmarkDotNet/blob/63e28c100a42a6492d09a0b93a8a4c141061bb0d/src/BenchmarkDotNet/Extensions/ProcessExtensions.cs#L138

https://github.com/dotnet/BenchmarkDotNet/blob/63e28c100a42a6492d09a0b93a8a4c141061bb0d/src/BenchmarkDotNet/Extensions/ProcessExtensions.cs#L23

so the extra sleep should not be needed

```suggestion
```